src/main/java/com/example/springboot/helper/FileClassifierUtil.java

```diff
@@ -68,8 +68,21 @@ public static String classifyForm(String file) {
 				String key = entry.getKey();
 				List<String> value = new ArrayList<String>(entry.getValue());
 				List<String> extractedTextList = new ArrayList<String>(processedTextList);
-				extractedTextList.retainAll(value);
-				matchingKeyWords.put(key, extractedTextList.size());
+				//extractedTextList.retainAll(value);
+				//matchingKeyWords.put(key, extractedTextList.size());
+				int count = 0;
+				for(int i=0; i<value.size(); i++) {
+					for(int j=0; j<extractedTextList.size(); j++) {
+						String keywordsValue = value.get(i).toString();
+						String extractedTextValue = extractedTextList.get(j).toString();
+						if (!extractedTextValue.equals("") && !extractedTextValue.equals(null) && extractedTextValue.length() > 0) {
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [EC: Call to equals(null) (EC_NULL_ARG)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ec-call-to-equals-null-ec-null-arg).
This method calls equals(Object), passing a null value as the argument. According to the contract of the equals() method, this call should always return false.